python/ray/data/dataset.py

```diff
@@ -2034,8 +2038,7 @@ def sort(
     def zip(self, other: "Dataset[U]") -> "Dataset[(T, U)]":
         """Zip this dataset with the elements of another.
 
-        The datasets must have identical num rows, block types, and block sizes,
-        e.g. one was produced from a :meth:`~.map` of another. For Arrow
+        The datasets must have the same number of rows. For Arrow
```

Comment:
we should document the semantics of aligning num of rows in blocks, and hence the implied repartitioning.